gluten-substrait/src/main/scala/org/apache/gluten/execution/ScanTransformerFactory.scala

```diff
@@ -55,6 +56,11 @@ object ScanTransformerFactory {
           scanExec.tableIdentifier,
           scanExec.disableBucketedScan
         )
+        if (!transform.doValidate().ok()) {
+          transform.copy(dataFilters = PushDownUtil.pushFilters(scanExec.dataFilters))
+        } else {
+          transform
+        }
```

Comment:
The code in `ScanTransformerFactory` is used by validator and offload rules. It feels a little weird to do validation in it? Do we have better choices? If not it's also OK to me to keep code like this.

Reply:
How about use only `pushedFilter` here and rely on `PushDownFilterToScan` for subsequent pushdown?

Reply:
Sounds feasible to me. Thanks.